src/pages/SignIn/index.js

```diff
@@ -1,20 +1,55 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 
 import Button from '@material-ui/core/Button';
 import Checkbox from '@material-ui/core/Checkbox';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { useLocation } from 'react-router-dom';
+
+import queryString from '@/package/queryString';
 
 import antiqueUniverseLogo from '../../assets/antique-universe-logo.png';
 
+import KaKaoOauthStrategy from './KaKaoOauthStrategy';
+import OauthStrategy from './OauthStrategy';
+
 import './signin.css';
 
+const KAKAO_VENDOR = 'KAKAO';
+
 const SignIn = () => {
+  const location = useLocation();
   const [staySigninState, setStaySigninState] = useState(false);
+  const oauthStrategy = useRef(new OauthStrategy());
+  const { code } = queryString.parse(location.search);
+  const vendor = window.localStorage.getItem('vendor');
 
-  const clickKakaoSigninBtn = () => alert('Kakao signin');
-  const clickNaverSigninBtn = () => alert('Kakao signin');
+  useEffect(() => {
+    if (vendor === KAKAO_VENDOR) {
+      oauthStrategy.current = new KaKaoOauthStrategy();
+    } else {
+      oauthStrategy.current = new OauthStrategy();
+    }
+  }, [vendor]);
 
-  const clickStaySigninCheckbox = () => setStaySigninState(!staySigninState);
+  useEffect(() => {
+    if (code) {
+      (async () => {
+        const token = await oauthStrategy.current.requestToken(code);
+        console.log(token);
+      })();
+    }
+  }, [code]);
+
+  const clickSigninBtn = () => {
+    oauthStrategy.current.requestCode();
+  };
+  const clickKakaoSigninBtn = () => {
+    window.localStorage.setItem('vendor', KAKAO_VENDOR);
```

Comment:
위에서 설명한 oauth 서비스 콜백 이슈가 없었더라면 그냥 다음과 같은 코드가 되었을 것 같습니다.. ㅠㅠ
```suggestion
    oauthStrategy.current = new KaKaoOauthStrategy();
```